src/Microsoft.OData.Client/BaseEntityType.cs

```diff
@@ -6,12 +6,17 @@
 
 namespace Microsoft.OData.Client
 {
-    /// <summary>
-    /// Base type of entity type to include <see cref="T:Microsoft.OData.Client.DataServiceContext" /> for function and action invocation
-    /// </summary>
-    public class BaseEntityType
-    {
-        /// <summary>DataServiceContext for query provider</summary>
-        protected internal DataServiceContext Context { get; set; }
-    }
+	/// <summary>
+	/// Base type of entity type to include <see cref="T:Microsoft.OData.Client.DataServiceContext" /> for function and action invocation
+	/// </summary>
+	public class BaseEntityType : IBaseEntityType
+	{
+		/// <inheritdoc/>
+		protected internal DataServiceContext Context { get; set; }

```

Comment:
Can't this be simplified to just `public DataServiceContext Context { get; set; }` instead of the explicit interface implementation?